syntax = "proto3";

package artifact.artifact.v1alpha;

// Artifact definitions
import "artifact/artifact/v1alpha/chunk.proto";


// QuestionAnsweringRequest
message QuestionAnsweringRequest {
    // id of the namespace
    string namespace_id = 1;
    // id of the catalog
    string catalog_id = 2;
    // question to be answered
    string question = 3;
    // topk default to 5
    int32 topk = 4;
  }
  
// QuestionAnsweringResponse
message QuestionAnsweringResponse {
// answer to the question
string answer = 1;
// chunks
repeated SimilarityChunk similarChunks = 2;
}

